Archives/ClmFile.cpp

```diff
@@ -134,41 +98,32 @@ namespace Archives
 
 		try
 		{
-			FileStreamWriter fileStreamWriter(pathOut);
+			FileStreamWriter waveFileWriter(pathOut);
 
-			fileStreamWriter.Write((char*)&header, sizeof(header));
+			waveFileWriter.Write((char*)&header, sizeof(header));
```

Comment:
I like the template idea and will investigate implementing it. This reminds me of generics in C#. 

I'm not sure of the pros and cons of using char* vs void* for the buffer. If it seems more correct to you, let's switch to void* but I would prefer switching implementations from char* to void* in the master branch as it will cause a large number of code changes that are unrelated to this branch.

Reply:
Good call. Let's make these changes in another branch.